packages/compass-smoke-tests/src/tests/auto-update-to.ts

```diff
@@ -0,0 +1,78 @@
+import assert from 'node:assert/strict';
+import { executeAsync } from '../execute';
+import { startAutoUpdateServer } from './update-server';
+import { type Channel } from '../build-info';
+
+type RunUpdateToTestOptions = {
+  appName: string;
+  appPath: string;
+  autoUpdatable?: boolean;
+  bucketKeyPrefix: string;
+  version: string;
+  channel: Channel;
+};
+
+export async function testAutoUpdateTo({
+  appName,
+  appPath,
+  autoUpdatable,
+  channel,
+  version,
+  bucketKeyPrefix,
+}: RunUpdateToTestOptions) {
+  process.env.PORT = '0'; // dynamic port
```

Comment:
Ideally, we would be able to pass this through the default function exported by the update server package, which would then be passed to the `UpdateChecker` constructor:

https://github.com/10gen/compass-mongodb-com/blob/a1fb99908815d3c3ab0efb5960430cc3faf99a15/src/update-checker.js#L40

Reply:
yeah it is a bit janky at the moment. That's not where the port is even used, though. update-checker is not the server itself. I think we can make a PR to compass-mongodb-com to allow us to clean it up, then another one here to use that. I'd do it as a follow-up, though, so we can have the tests running in the meantime.